pkg/products/threescale/reconciler.go

```diff
@@ -3308,50 +3290,58 @@ func (r *Reconciler) findCustomDomainCr(ctx context.Context, serverClient k8scli
 	return integreatlyv1alpha1.PhaseFailed, fmt.Errorf("finding CustomDomain CR failed: %v", err)
 }
 
-func (r *Reconciler) ping3scalePortals(ctx context.Context, serverClient k8sclient.Client, ips []net.IP) (integreatlyv1alpha1.StatusPhase, error) {
+func (r *Reconciler) ping3scalePortals(ctx context.Context, serverClient k8sclient.Client) (integreatlyv1alpha1.StatusPhase, error) {
 	format := "failed to retrieve %s 3scale route"
 	portals := map[string]metrics.PortalInfo{}
+	ips := make(map[string][]net.IP)
+	metricsLabelMap := make(map[string]string)
+	metricsLabelMap[labelRouteToSystemDeveloper] = metrics.LabelSystemDeveloper
+	metricsLabelMap[labelRouteToSystemProvider] = metrics.LabelSystemProvider
+	metricsLabelMap[labelRouteToSystemMaster] = metrics.LabelSystemMaster
+
+	for _, threeScaleService := range [3]string{labelRouteToSystemDeveloper, labelRouteToSystemProvider, labelRouteToSystemMaster} {
+		var prefix string
+		if threeScaleService == labelRouteToSystemDeveloper {
+			prefix = "3scale."
+		} else if threeScaleService == labelRouteToSystemProvider {
+			prefix = "3scale-admin."
+		}
+		route, err := r.getThreescaleRoute(ctx, serverClient, threeScaleService, func(r routev1.Route) bool {
+			return strings.HasPrefix(r.Spec.Host, prefix)
+		})
+		if err != nil || route == nil {
+			return integreatlyv1alpha1.PhaseFailed, fmt.Errorf(format, threeScaleService)
+		}
+		ingressRouters := []string{"router-default"}
+		if len(route.Status.Ingress) > 1 {
```

Comment:
I don't think we can assume that the first route will be the default one. On the latest OHSS customer issue this is not the case. The 3scale route only has one ingress on it and it's not the default one.

Reply:
before write it i looked at 3scale, and seems "router-default" is always present .. But I will look how to handle it

Reply:
done, as suggested:  https://github.com/integr8ly/integreatly-operator/pull/3338